packages/cli-hermes/src/profileHermes/downloadProfile.ts

```diff
@@ -0,0 +1,175 @@
+import {Config} from '@react-native-community/cli-types';
+import {execSync} from 'child_process';
+import {logger, CLIError} from '@react-native-community/cli-tools';
+import chalk from 'chalk';
+import fs from 'fs';
+import path from 'path';
+import os from 'os';
+import transformer from 'hermes-profile-transformer';
+import {findSourcemap, generateSourcemap} from './sourcemapUtils';
+/**
+ * Get the last modified hermes profile
+ * @param packageName
+ */
+function getLatestFile(packageName: string): string {
+  try {
+    const file = execSync(`adb shell run-as ${packageName} ls cache/ -tp | grep -v /$ | egrep '\.cpuprofile' | head -1
+        `);
+    return file.toString().trim();
+  } catch (e) {
+    throw new Error(e);
+  }
+}
+
+function execSyncWithLog(command: string) {
+  if (logger.isVerbose()) {
+    logger.debug(`${command}\n`);
+  }
+
+  return execSync(command);
+}
+
+/**
+ * Get the package name of the running React Native app
+ * @param config
+ */
+function getPackageName(config: Config) {
+  const androidProject = config.project.android;
+
+  if (!androidProject) {
+    throw new CLIError(`
+  Android project not found. Are you sure this is a React Native project?
+  If your Android files are located in a non-standard location (e.g. not inside \'android\' folder), consider setting
+  \`project.android.sourceDir\` option to point to a new location.
+`);
+  }
+  const {manifestPath} = androidProject;
+  const androidManifest = fs.readFileSync(manifestPath, 'utf8');
+
+  let packageNameMatchArray = androidManifest.match(/package="(.+?)"/);
+  if (!packageNameMatchArray || packageNameMatchArray.length === 0) {
+    throw new CLIError(
+      'Failed to build the app: No package name found. Found errors in /src/main/AndroidManifest.xml',
+    );
+  }
+
+  let packageName = packageNameMatchArray[1];
+
+  if (!validatePackageName(packageName)) {
+    logger.warn(
+      `Invalid application's package name "${chalk.bgRed(
+        packageName,
+      )}" in 'AndroidManifest.xml'. Read guidelines for setting the package name here: ${chalk.underline.dim(
+        'https://developer.android.com/studio/build/application-id',
+      )}`,
+    );
+  }
+  return packageName;
+}
+
+/**
+ * Validates that the package name is correct
+ * @param packageName
+ */
+function validatePackageName(packageName: string) {
+  return /^[a-z][a-z0-9_]*(\.[a-z][a-z0-9_]*)+$/.test(packageName);
+}
```

Comment:
Same: https://github.com/react-native-community/cli/blob/f497f5a9f1bb0980488b989034d96b30ffd08747/packages/platform-android/src/commands/runAndroid/index.ts#L27-L29

I would create src/utils inside `platform-android` package and export them, so that we can use them here. What do you think?

Reply:
Yeah I think this is a good idea. I'll go ahead with that 🚀